src/NATS.Client.Core/INatsConnection.cs

```diff
@@ -17,10 +17,12 @@ public interface INatsConnection
     /// </summary>
     /// <param name="subject">The destination subject to publish to.</param>
     /// <param name="payload">The message payload data.</param>
+    /// <param name="replyTo">Optional reply-to subject.</param>
+    /// <param name="headers">Optional message headers.</param>
     /// <param name="opts">A <see cref="NatsPubOpts"/> for publishing options.</param>
     /// <param name="cancellationToken">A <see cref="CancellationToken"/> used to cancel the command.</param>
     /// <returns>A <see cref="ValueTask"/> that represents the asynchronous send operation.</returns>
-    ValueTask PublishAsync(string subject, ReadOnlySequence<byte> payload = default, NatsPubOpts? opts = default, CancellationToken cancellationToken = default);
+    ValueTask PublishAsync(string subject, ReadOnlySequence<byte> payload = default, string? replyTo = default, NatsHeaders? headers = default, NatsPubOpts? opts = default, CancellationToken cancellationToken = default);
```

Comment:
do you think there is a benefit to having `headers` appear before `replyTo`, as it is likely a more common option than `replyTo` - for those who call with a positional syntax

I can also see the argument that `replyTo` should come first, because it was defined in the protocol before Headers were

I don't have a strong preference but thought it was worth bringing up

Reply:
This is a good point. Like you said, I was thinking of the protocol order but that shouldn't be a concern here. Usability is more important and I agree headers are more likely to be used as metadata than reply-to.

Reply:
Fixed. Also swapped them for `NatsMsg.ReplyAsync()` methods.